src/Utilities/AxiosBatchTransport.ts

```diff
@@ -0,0 +1,113 @@
+import { AxiosInstance } from 'axios';
+import Transport from 'winston-transport';
+import _ from 'lodash';
+import { getUserId } from '../Enums/UserRole';
+
+interface AxiosBatchTransportOptions extends Transport.TransportStreamOptions {
+    axios: AxiosInstance;
+    throttleTimeInMillis?: number;
+    loggingEndpoint: string;
+}
+
+interface LogMessage {
+    message: string;
+    time: Date;
+    level: string;
+    meta?: unknown;
+}
+
+export default class AxiosBatchTransport extends Transport {
```

Comment:
My only concern with this is performance. Can the batching process run in the background? Does switching tabs affect the batching?

Reply:
I don't understand what you mean? It should be scoped to a tab and the batching process is just putting it into an array and sending them all at once

Technically I guess if you had two tabs open you could send twice as many requests but still I don't think that is a big deal